### PR TITLE
[WIP] Add support for an init container in simple-app

### PR DESCRIPTION
**Pull request by mrterry**
Add Values for an init_container

### CODE REVIEW

**Comment by mrterry** [comment]
IMO the interface for this change is much more important that the details of the resources created.  @diranged @smitelli can I get a directional review of this before I start creating templates?

I don't think the init container needs a lot of configuration.  It needs env vars, volumes, and resources for sure, but I'm not sure needs much more. I don't need network access, but users might.  As such all things network are unimplemented in the interest of YAGNI.

As implemented, all the env vars, volumes, resources are entirely decoupled from the "main" container. We should consider having the init_container and main_container share configuration for env_vars, creds, volumes. I can't quite tell if this is a genius idea that will simplify configuration or a terrible idea that will turn into a mountain of shared passwords and mess.

File: charts/simple-app/values.yaml [hunk 1]
```diff
@@ -122,6 +122,46 @@ readinessProbe:
     path: /
     port: http
 
+# An optional init container. This container starts and exits before the
```
Comment (diranged) [inline charts/simple-app/values.yaml:125]: So... I definitely don't want to create a dedicated configuration here for the initContainer. I would consier a single large parameter (list of maps)) that _is_ the `initContainers` value. The thing is that once you have one initContainer, who is to say you won't have two, or three... there's just too much customization available at that point.

I'd think:

```
# values.yaml
initContainers: []
```

```
# deployment.yaml
apiVersion: v1
...
spec:
  template:
    {{- with .Values.initContainers }}
    {{- tpl (toYaml .) $ | nindent XX }}
    {{- end }}
...
```

Even this though you should prototype on your laptop in your own chart and just see how it feels

Comment (mrterry) [reply charts/simple-app/values.yaml:125]: interesting.  as implemented, this has one bespoke initContainer, whose configuration is limited by the bits exposed in `values.yaml`.  This sounds like you would prefer a blind import from `values.yaml` with unlimited configuration.  Blindly passing `initContainers` through to the pod spec is certainly more powerful, but it is also a lot more rope for the user to hang themselves with.

Comment (diranged) [reply charts/simple-app/values.yaml:125]: yup... I feel like the moment we're talking about something as complex and powerful as an initContainer, we should release the reigns.

Comment (mrterry) [reply charts/simple-app/values.yaml:125]: ok.  see #173 for that approach.

**Pull request closed without merging**